### PR TITLE
Enable TLS in filebeat

### PR DESCRIPTION
**Pull request by VariableDeclared**
Add TLS Support to the filebeat charm

[Depends on PR for elastic beats layer](https://github.com/juju-solutions/layer-beats-base/pull/42)

Thanks,
Peter

### CODE REVIEW

**Review by agileshaw — COMMENTED** [review]
Overall LGTM. Just one question about the build image change

File: charmcraft.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ parts:
 bases:
   - build-on:
     - name: ubuntu
-      channel: "20.04"
+      channel: "22.04"
```
[the remark this answers is not in the document]
Comment (VariableDeclared) [reply charmcraft.yaml:18]: yes, I can change this back - it was a issue with my workstation I was working around :-)

[the remark this answers is not in the document]
Comment (agileshaw) [reply charmcraft.yaml:18]: Could you please attach the output for the 20.04 build?

[the remark this answers is not in the document]
Comment (VariableDeclared) [reply charmcraft.yaml:18]: sure: https://pastebin.ubuntu.com/p/YVbrhyNRmB/

**Comment by VariableDeclared** [comment]
hey @agileshaw any advice on the focal/jammy build issue?

**Comment by agileshaw** [comment]
Hi @VariableDeclared . Apologize for the late reply. 

This is a known issue with Jinja and MarkupSafe2 in focal.  A workaround can be found here: https://github.com/juju/charm-tools/pull/650

**Comment by agileshaw** [comment]
Since the issue only seems to appear on focal, I think we can use jammy as build base (so you don't need to change back the metadata.yaml). Please attach the log of a jammy-built charm and test its behavior with some simple bundle (e.g. https://git.launchpad.net/charm-graylog/tree/src/tests/functional/tests/bundles/base-graylog.yaml). If everything checks out, that's a +1 from my book. 

I'm also adding @esunar as a reviewer.

**Comment by lathiat** [comment]
The fix here may work:
https://discourse.charmhub.io/t/install-or-update-python-packages-before-packing-a-charm/5158
https://github.com/canonical/charmcraft/issues/551

```
parts:
  charm:
    charm-python-packages:
      # https://github.com/canonical/charmcraft/issues/551
      - setuptools
```

**Comment by VariableDeclared** [comment]
Following up on my last comment - the issue is related to the `charm` plugin and `reactive` plugin. In this case we're using reactive which doesn't appear to support this arguement.

**Review by esunar — APPROVED** [review]

File: metadata.yaml [hunk 2]
```diff
@@ -6,4 +6,17 @@ description: |
...
+requires:
+  beats-host:
+    interface: juju-info
+    scope: container
+  logstash:
+    interface: elastic-beats
+  elasticsearch:
+    interface: elasticsearch
+  kafka:
+    interface: kafka
```
Comment (VariableDeclared) [inline metadata.yaml:22]: hey @esunar and @agileshaw I've pushed some changes which I hope will help with the issues seen. Seems that build no longer works on my workstation, I had to add

Comment (VariableDeclared) [inline metadata.yaml:22]: hey @esunar and @agileshaw I've pushed some changes which I hope will help with the issues seen. Seems that build no longer works on my workstation, I had to add these lines to have a successful build.

Comment (agileshaw) [reply metadata.yaml:22]: The changes looks good to me. Could you please attach the log from some simple bundle deployment? (e.g. https://git.launchpad.net/charm-graylog/tree/src/tests/functional/tests/bundles/base-graylog.yaml). I would like to see all if this jammy-built charm doesn't break such a deployment.

Comment (VariableDeclared) [reply metadata.yaml:22]: hey sure, this is the bundle I am using today:

https://pastebin.canonical.com/p/kcddqf7WsQ/

Comment (agileshaw) [reply metadata.yaml:22]: And do you have a `juju status` (or something similar that shows deployment status) output of the deployed bundle?

Comment (VariableDeclared) [reply metadata.yaml:22]: sure, juju status is here: https://pastebin.canonical.com/p/jFRJTx3x7p/

**Pull request merged**